nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_Discovery.c

```diff
@@ -1442,6 +1442,24 @@ static NFCSTATUS phNciNfc_DeActvNtfCb(void *pContext,void *pInfo,NFCSTATUS wStat
                         wDeActvStatus = NFCSTATUS_SUCCESS;
                     }
                 }
+                else if (pTransInfo->pbuffer[1] == (uint8_t)phNciNfc_e_DhRequestFailed)
+                {
+                    if ((pNciContext->NciDiscContext.eDeActvType == phNciNfc_e_SleepMode) && \
+                        (pTransInfo->pbuffer[0] == phNciNfc_e_SleepMode))
+                    {
+                        wDeActvStatus = NFCSTATUS_SUCCESS;
+                    }
+                    if ((pNciContext->NciDiscContext.eDeActvType == phNciNfc_e_SleepAfMode) && \
+                        (pTransInfo->pbuffer[0] == phNciNfc_e_SleepAfMode))
+                    {
+                        wDeActvStatus = NFCSTATUS_SUCCESS;
+                    }
+                    else if ((pNciContext->NciDiscContext.eDeActvType == phNciNfc_e_IdleMode) && \
+                             (pTransInfo->pbuffer[0] == phNciNfc_e_IdleMode))
```

Comment:
Those new conditions are checking the following statement from the NCI2.0 specification chapter 5.2.5 State RFST_POLL_ACTIVE:

"In this state the DH MAY send RF_DEACTIVATE_CMD (Sleep Mode or Sleep_AF Mode) to
deactivate communication with the Remote NFC Endpoint. The NFCC SHALL send
RF_DEACTIVATE_RSP with STATUS_OK. Depending on the activated interface and protocol,
the NFCC might need to issue RF commands to put the Remote NFC Endpoint into sleep state. If
there was an error in executing the protocol deactivation procedures defined by the RF Interface,
the NFCC SHALL notify the DH by sending RF_DEACTIVATE_NTF (Sleep Mode or
Sleep_AF Mode, DH request failed due to error), and the state will remain
RFST_POLL_ACTIVE."

and

"In this state the DH MAY send RF_DEACTIVATE_CMD (Idle Mode) to deactivate
communication with the Remote NFC Endpoint and stop RF Discovery. The NFCC SHALL send
RF_DEACTIVATE_RSP with STATUS_OK and, depending on the activated interface and
protocol, might issue RF commands to deactivate the Remote NFC Endpoint. If there was an
error in executing the protocol deactivation procedures defined by the RF Interface, the NFCC
SHALL notify the DH by sending RF_DEACTIVATE_NTF (Idle Mode, DH request failed due
to error)."

pNciContext->NciDiscContext.eDeActvType is the deactivation type saved during phNciNfc_Deactivate in nfc\libs\NfcCoreLib\lib\Nci\phNciNfc.c (RF_DEACTIVATE_CMD).

Would you like to have an update with an additional comment in the code or in the commit message ?